services/brig/test/integration/Util.hs

```diff
@@ -1118,3 +1127,72 @@ withMockedFederatorAndGalley opts _domain fedResp galleyHandler action = do
   where
     combineResults :: ((a, [Mock.FederatedRequest]), [ReceivedRequest]) -> (a, [Mock.FederatedRequest], [ReceivedRequest])
     combineResults ((a, mrr), rr) = (a, mrr, rr)
+
+newtype WaiTestFedClient a = WaiTestFedClient {unWaiTestFedClient :: ReaderT Domain WaiTest.Session a}
+  deriving (Functor, Applicative, Monad, MonadIO)
+
+instance Servant.RunClient WaiTestFedClient where
+  runRequestAcceptStatus expectedStatuses servantRequest = WaiTestFedClient $ do
+    domain <- ask
+    let req' = fromServantRequest domain servantRequest
+    res <- lift $ WaiTest.srequest req'
+    let servantResponse = toServantResponse res
+    let status = Servant.responseStatusCode servantResponse
+    let statusIsSuccess =
+          case expectedStatuses of
+            Nothing -> HTTP.statusIsSuccessful status
+            Just ex -> status `elem` ex
+    unless statusIsSuccess $
+      unWaiTestFedClient $ throwClientError (FailureResponse (bimap (const ()) (\x -> (Servant.BaseUrl Servant.Http "" 80 "", cs (toLazyByteString x))) servantRequest) servantResponse)
+    pure servantResponse
+  throwClientError = liftIO . throw
+
+fromServantRequest :: Domain -> Servant.Request -> WaiTest.SRequest
+fromServantRequest domain r =
+  let pathBS = "/federation" <> Data.String.Conversions.cs (toLazyByteString (Servant.requestPath r))
+      bodyBS = case Servant.requestBody r of
+        Nothing -> ""
+        Just (bdy, _) -> case bdy of
+          Servant.RequestBodyLBS lbs -> Data.String.Conversions.cs lbs
+          Servant.RequestBodyBS bs -> bs
+          Servant.RequestBodySource _ -> error "fromServantRequest: not implemented for RequestBodySource"
+      req =
+        Wai.defaultRequest
+          { Wai.requestMethod = Servant.requestMethod r,
+            Wai.rawPathInfo = pathBS,
+            Wai.rawQueryString = renderQuery True (toList (Servant.requestQueryString r)),
+            Wai.requestHeaders =
+              (("Content-Type",) . renderHeader . snd <$> maybeToList (Servant.requestBody r))
+                <> ((\x -> ("Accept", renderHeader x)) <$> toList (Servant.requestAccept r))
```

Comment:
This could lead to a list of `Accept` headers. What we likely want is an `Accept` header with a list of media types.

Reply:
This is fixed, now.